communities/rpc/src/lib.rs

```diff
@@ -98,7 +98,7 @@ macro_rules! refresh_cache {
         let mut cid_names: Vec<CidName> = vec![];
 
         cids.iter().for_each(|cid| {
-            $self.get_storage(cid.as_ref()).map_or_else(
+            $self.get_storage(&cid.as_array()).map_or_else(
```

Comment:
Here you should be able to simply call the borrow `&cid`.

Anyhow, @brenzi can you explain to me, why you used the macro?

Was it to circumvent setting an extensive amount of trait bounds?

Reply:
With the macro we have less trait bounds on the fundamental Struct so test mocking is easier
tl;dr: I couldn't make it work otherwise

Reply:
couldn't either.. there are ways to just use the memory area of the struct itself with unsafe code, but seeing the size of the method and expressivity, I prefer this solution